cmake/CompilerChecks.cmake

```diff
@@ -11,6 +11,13 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
     endif()
 endif()
 
+# MSVC automatically defines -D_DEBUG when /MTd or /MDd is set, so we
+# need to make sure it gets added for other compilers too
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_DEBUG")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_DEBUG")
+endif()
```

Comment:
Under what cases would this not be set? Wouldn't this be controlled by the CMake Build Type (Debug, Release, RelWithDebInfo, etc.)?